Source/Dafny/Compilers/Compiler-python.cs

```diff
@@ -0,0 +1,598 @@
+using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Diagnostics;
+using System.Diagnostics.Contracts;
+using System.IO;
+using Microsoft.Boogie;
+
+
+namespace Microsoft.Dafny {
+  public class PythonCompiler : Compiler {
+    public PythonCompiler(ErrorReporter reporter) : base(reporter) {
+    }
+
+    public override string TargetLanguage => "Python";
+
+    const string DafnySetClass = "_dafny.Set";
+    const string DafnyMultiSetClass = "_dafny.MultiSet";
+    const string DafnySeqClass = "_dafny.Seq";
+    const string DafnyMapClass = "_dafny.Map";
+
+    public override void EmitCallToMain(Method mainMethod, string baseName, ConcreteSyntaxTree wr) {
+      Coverage.EmitSetup(wr);
+      wr.WriteLine("__default.Main()", mainMethod.EnclosingClass, mainMethod);
+    }
+
+    protected override ConcreteSyntaxTree CreateStaticMain(IClassWriter cw) {
+      var wr = (cw as PythonCompiler.ClassWriter).MethodWriter;
+      return wr.WriteLine("def Main():");
+    }
+
+    protected override ConcreteSyntaxTree CreateModule(string moduleName, bool isDefault, bool isExtern,
+      string libraryName,
+      ConcreteSyntaxTree wr) {
+      return wr;
+    }
+
+    protected override string GetHelperModuleName() {
+      throw new NotImplementedException();
+    }
+
+    protected override IClassWriter CreateClass(string moduleName, string name, bool isExtern, string fullPrintName,
+      List<TypeParameter> typeParameters,
+      TopLevelDecl cls, List<Type> superClasses, IToken tok, ConcreteSyntaxTree wr) {
+
+      var w = wr.WriteLine("class {0}:", name);
+
+      var methodWriter = w.NewBlock(open: BraceStyle.Pindent, close: BraceStyle.Pindent);
+      ConcreteSyntaxTree fieldWriter = w.NewBlock(open: BraceStyle.Pindent, close: BraceStyle.Pindent);
+      return new ClassWriter(this, methodWriter, fieldWriter);
+
+    }
+
+    protected override IClassWriter CreateTrait(string name, bool isExtern, List<TypeParameter> typeParameters,
+      List<Type> superClasses, IToken tok,
+      ConcreteSyntaxTree wr) {
+      throw new NotImplementedException();
+    }
+
+    protected override ConcreteSyntaxTree CreateIterator(IteratorDecl iter, ConcreteSyntaxTree wr) {
+      throw new NotImplementedException();
+    }
+
+    protected override IClassWriter DeclareDatatype(DatatypeDecl dt, ConcreteSyntaxTree wr) {
+
+      return null;
+
+
```

Comment:
It is called in the start and if instead of this I put 'NotImplementedException' , it will stop the execution. I have added if condition over here as other compiler and else part has the 'NotImplementedException' .

Reply:
It is called in start so has to let it return null back but I am adding if condition and else part will throw the exception.